### PR TITLE
Handle Images with RGBA Colors (transparency)

### PR DESCRIPTION
**Pull request by bunnymatic**
* Add handling for pngs with transparency
* stringify keys from histogram to be consistent

### CODE REVIEW

**Comment by bunnymatic** [comment]
This should fix #58 .

File: lib/mogrify.ex [hunk 1]
```diff
@@ -77,21 +78,21 @@ defmodule Mogrify do
...
+  defp clean_histogram_entry( { k, "" } ), do: { k |> String.to_atom, 0 }
+  defp clean_histogram_entry( { k, v } ), do: { k |> String.to_atom, (v |> String.to_integer) }
+
   defp extract_histogram_data(entry) do
-    ~r/^\s+(?<count>\d+):\s+\((?<red>[\d\s]+),(?<green>[\d\s]+),(?<blue>[\d\s]+)\)\s+(?<hex>\#[abcdef\d]{6})\s+/
-    |> Regex.named_captures(entry |> String.downcase)
-    |> histogram_integerify
+    ~r/^\s+(?<count>\d+):\s+\((?<red>[\d\s]+),(?<green>[\d\s]+),(?<blue>[\d\s]+)(,(?<alpha>[\d\s]+))?\)\s+(?<hex>\#[abcdef\d]{6,8})\s+/i
+    |> Regex.named_captures(entry)
+    |> Enum.map( fn {k,v} -> { k, v |> String.trim } end )
```
Comment (talklittle) [inline lib/mogrify.ex]: You can use `Compat.string_trim/1` for Elixir 1.2 compatibility

File: lib/mogrify.ex [hunk 2]
```diff
@@ -54,12 +54,13 @@ defmodule Mogrify do
 
   iex> open("test/fixtures/rbgw.png") |> histogram
   [
-    %{"blue" => 255, "count" => 400, "green" => 0, "hex" => "#0000ff", "red" => 0},
-    %{"blue" => 0, "count" => 225, "green" => 255, "hex" => "#00ff00", "red" => 0},
-    %{"blue" => 0, "count" => 525, "green" => 0, "hex" => "#ff0000", "red" => 255},
-    %{"blue" => 255, "count" => 1350, "green" => 255, "hex" => "#ffffff", "red" => 255}
+    %{"alpha" => 0, "blue" => 255, "count" => 400, "green" => 0, "hex" => "#0000ff", "red" => 0},
```
Comment (talklittle) [inline lib/mogrify.ex]: I think it makes sense for `alpha` to default to 255 rather than 0 if it's RGB data.

File: mix.lock [hunk 3]
```diff
@@ -1,4 +1,2 @@
-%{
-  "earmark": {:hex, :earmark, "1.2.5", "4d21980d5d2862a2e13ec3c49ad9ad783ffc7ca5769cf6ff891a4553fbaae761", [:mix], [], "hexpm"},
-  "ex_doc": {:hex, :ex_doc, "0.18.3", "f4b0e4a2ec6f333dccf761838a4b253d75e11f714b85ae271c9ae361367897b7", [:mix], [{:earmark, "~> 1.1", [hex: :earmark, repo: "hexpm", optional: false]}], "hexpm"},
-}
+%{"earmark": {:hex, :earmark, "1.2.3", "206eb2e2ac1a794aa5256f3982de7a76bf4579ff91cb28d0e17ea2c9491e46a4", [:mix], []},
+  "ex_doc": {:hex, :ex_doc, "0.16.2", "3b3e210ebcd85a7c76b4e73f85c5640c011d2a0b2f06dcdf5acdb2ae904e5084", [:mix], [{:earmark, "~> 1.1", [hex: :earmark, optional: false]}]}}
```
Comment (talklittle) [inline mix.lock]: These `mix.lock` changes are unnecessary

File: lib/mogrify.ex [hunk 4]
```diff
@@ -310,4 +311,27 @@ defmodule Mogrify do
...
+
+  defp stringify_keys(map = %{}) do
+    map
+    |> Enum.map(fn {k, v} -> {Atom.to_string(k), stringify_keys(v)} end)
+    |> Enum.into(%{})
+  end
+
+  # Walk the list and stringify the keys of
+  # of any map members
+  defp stringify_keys([head | rest]) do
```
Comment (talklittle) [inline lib/mogrify.ex]: Style-wise I prefer

    defp stringify_keys(list) when is_list(list) do
      Enum.map(list, stringify_keys/1)
    end

**Comment by bunnymatic** [comment]
cool... i'll update.  I'm definitely a bit green in elixir so the style comments are appreciated.

Also... i noticed the tests are failing but they are around `pango` which were also failing on master so i didn't worry too much about it.

**Comment by talklittle** [comment]
Great, thanks for taking a look. Sounds like `pango` tests are failing for you locally, but it's okay because they appear to be passing on Travis.

The only Travis tests failing are due to `String.trim` on Elixir 1.2 because back then it was called `String.strip`. `Mogrify.Compat.string_trim` is a workaround for that.

**Comment by talklittle** [comment]
Thanks a lot for the patch! Mogrify 0.6.1 contains the fix.

**Pull request merged**